src/posix/Makefile.am

```diff
@@ -140,9 +140,11 @@ ot_daemon_SOURCES                                                      = \
 
 ot_daemon_LDADD                                                        = \
     $(top_builddir)/src/cli/libopenthread-cli-ftd.a                      \
+    $(top_builddir)/src/ncp/libopenthread-ncp-ftd.a                      \
```

Comment:
Is it possible to make spinel.c and hdlc.cpp into libopenthread-spinel.a which can be shared?

Reply:
Very good idea.. I suggest we do this in a later PR.

My idea for this change is to be able to share more code between posix-app and ncp module (e.g., ncp frame buffer and/or encoder/decoder, etc).  For now having the entire ncp-ftd library would be useful but later we can add all common/shared modules into a separate library.

Reply:
I didn't know ot-cli and ot-daemon needs the entire ncp-ftd library. Could you please share more details why it's useful to having the entire ncp-ftd library? I think this would increase sizes of ot-cli and ot-daemon.

Reply:
Verified locally that linking to ncp-ftd will not increase code size.

Reply:
Thanks @bukepo . We can include the entire lib for now (later we can create a separate lib). I expect the linkers to optimize out and remove all unused methods/definitions.